roles/openshift_health_checker/callback_plugins/zz_failure_summary.py

```diff
@@ -66,25 +66,37 @@ def _print_failure_summary(self):
 # Status: permanently disabled unless Ansible's API changes.
 # pylint: disable=protected-access
 def _format_failure(failure):
-    '''Return a list of pretty-formatted lines describing a failure, including
-    relevant information about it. Line separators are not included.'''
+    '''Return a list of pretty-formatted text entries describing a failure, including
+    relevant information about it. Line breaks can be included but not trailing newlines.'''
     result = failure['result']
     host = result._host.get_name()
     play = _get_play(result._task)
     if play:
         play = play.get_name()
     task = result._task.get_name()
     msg = result._result.get('msg', u'???')
-    rows = (
+    fields = (
         (u'Host', host),
         (u'Play', play),
         (u'Task', task),
         (u'Message', stringc(msg, C.COLOR_ERROR)),
     )
     if 'checks' in result._result:
-        rows += ((u'Details', stringc(pformat(result._result['checks']), C.COLOR_ERROR)),)
+        fields += ((u'Details', _format_failed_checks(result._result['checks'])),)
     row_format = '{:10}{}'
-    return [row_format.format(header + u':', body) for header, body in rows]
+    return [row_format.format(header + u':', body) for header, body in fields]
+
+def _format_failed_checks(checks):
+    '''Return pretty-formatted text describing checks that failed.'''
+    failed_check_msgs = []
+    for check, body in checks.items():
+        if body.get('failed', False):   # only show the failed checks
+            msg = body.get('msg', u"Failed without returning a message")
+            failed_check_msgs.append('check "%s":\n%s' % (check, msg))
+    if failed_check_msgs:
+        return stringc("\n\n".join(failed_check_msgs), C.COLOR_ERROR)
+    else:    # something failed but no checks will admit to it, so dump everything
+        return stringc(pformat(result._result['checks']), C.COLOR_ERROR)
```

Comment:
Static analysis also catch that, I think both flake8 and pylint should point at it.